test/fixtures.py

```diff
@@ -317,21 +353,153 @@ def open(self):
             tries += 1
         else:
             raise Exception('Failed to start KafkaInstance before max_timeout')
+
+        (self._client,) = self.get_clients(1, '_internal_client')
+
         self.out("Done!")
         self.running = True
-        atexit.register(self.close)
 
-    def __del__(self):
-        self.close()
```

Comment:
Why is it safe to remove this?

Reply:
My idea is that these should all be instantiated by pytest fixtures, and I'm handling the destruction in the fixture methods, as explained above.